cns/kubecontroller/nodenetworkconfig/conversion.go

```diff
@@ -95,6 +99,33 @@ func CreateNCRequestFromStaticNC(nc v1alpha.NetworkContainer) (*cns.CreateNetwor
 			NCVersion: int(nc.Version),
 		}
 	}
+
+	// Add IPs from CIDR block to the secondary IPConfigs
+	if nc.Type == v1alpha.VNETBlock {
+		// Delete the IPs upto(including) first IP of primary prefix
+		startAddr := primaryPrefix.Masked().Addr()
+		endAddr := primaryPrefix.Addr()
+		for addr := startAddr; addr != endAddr.Next() && primaryPrefix.Contains(addr); addr = addr.Next() {
+			delete(secondaryIPConfigs, addr.String())
+		}
+
+		for _, ipAssignment := range nc.IPAssignments {
+			cidrPrefix, err := netip.ParsePrefix(ipAssignment.IP)
+			if err != nil {
+				return nil, errors.Wrapf(ErrInvalidCIDRBlock, "CIDR Block: %s", ipAssignment.IP)
+			}
+
+			// iterate through all IP addresses in the CIDR block described by cidrPrefix and
+			// add them to the request as secondary IPConfigs.
+			for addr := cidrPrefix.Addr(); cidrPrefix.Contains(addr); addr = addr.Next() {
```

Comment:
should start with the zeroth address in these CIDRs also
```suggestion
			for addr := cidrPrefix.Masked().Addr(); cidrPrefix.Contains(addr); addr = addr.Next() {
```